py3status/modules/sysdata.py

```diff
@@ -126,6 +132,49 @@ def memory(self, unit='GiB'):
         # Otherwise, results are in gigabytes.
         return total_mem, used_mem, used_mem_p, unit
 
+    def swap(self, unit='GiB'):
+        """
+        Parse /proc/meminfo, grab the swap capacity and used size
+        then return; Swap size 'total_swap', Used_swap, percentage
+        of used spwap, and units of swap (KiB, MiB, GiB).
+        """
+
+        memi = {}
+        with open('/proc/meminfo', 'r') as fd:
+            for s in fd:
+                tok = s.split()
+                memi[tok[0]] = float(tok[1])
```

Comment:
We are already doing this in memory().  It would be better if we only parse this once and share the info.